imagescan/scanner.go

```diff
@@ -252,29 +257,53 @@ func (s *Scanner) ScanImage(ctx context.Context, params ScanImageParams) (rerr e
 			if err := jobs.Delete(ctx, jobSpec.Name, metav1.DeleteOptions{
 				PropagationPolicy: lo.ToPtr(metav1.DeletePropagationBackground),
 			}); err != nil && !apierrors.IsNotFound(err) {
-				rerr = err
+				rerr = fmt.Errorf("deleting finished job: %w", err)
 			}
 		}()
 	}
 
 	// If job already exist wait for completion and exit.
 	_, err := jobs.Get(ctx, jobSpec.Name, metav1.GetOptions{})
 	if err == nil {
-		return s.waitForCompletion(ctx, jobs, jobName)
+		if err := s.waitForCompletion(ctx, jobs, jobName); err != nil {
+			return fmt.Errorf("job already exist, wait for completion: %w", err)
+		}
+		return nil
 	}
 
 	// Create new job and wait for completion.
 	_, err = jobs.Create(ctx, jobSpec, metav1.CreateOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("creating job: %w", err)
 	}
 
 	if params.WaitForCompletion {
-		return s.waitForCompletion(ctx, jobs, jobName)
+		if err := s.waitForCompletion(ctx, jobs, jobName); err != nil {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			jobPod, _ := s.getJobPod(ctx, jobName)
+			if jobPod != nil {
+				conds := getPodConditionsString(jobPod.Status.Conditions)
+				return fmt.Errorf("wait for completion, pod_conditions=%s: %w", conds, err)
+			}
+			return fmt.Errorf("wait for completion: %w", err)
+		}
 	}
 	return nil
 }
 
+func getPodConditionsString(conditions []corev1.PodCondition) string {
+	var condStrings []string
+	for _, condition := range conditions {
+		reason := condition.Reason
+		if reason == "" {
+			reason = condition.Message
+		}
+		condStrings = append(condStrings, fmt.Sprintf("[type=%s, status=%s, reason=%s]", condition.Type, condition.Status, reason))
```

Comment:
is there any reason for such string format [*]? maybe json string will be better?

Reply:
json string in the middle of text message doesn't seem to be more practical